sdk/common/src/main/java/io/opentelemetry/sdk/resources/EntityDetector.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources;
+
+import java.util.List;
+
+/**
+ * The Entity detector in the SDK is responsible for detecting possible entities that could identify
+ * the SDK (called "associated entities"). For Example, if the SDK is running in a kubernetes pod,
+ * it may provide an Entity for that pod.
+ */
+public interface EntityDetector {
```

Comment:
What's the thinking behind making EntityDetector a part of the SDK, instead of part of `opentelemetry-sdk-extension-autoconfigure-spi` like [ResourceProvider](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ResourceProvider.java)?

Reply:
Long term direction is for ResourceProvider to become part of SDK and allow mutating the set of entities or descriptive attributes over time.

I.e. ResourceProvider might provide something *like* an observer-pattern for gaining access to the current definition of resource.  You can then pass this to `{Signal}Provider`.

Initially we don't need that and I expect `ResourceProvider` to be an internal detail.